### PR TITLE
Be more flexible on accepted paths

### PR DESCRIPTION
**Pull request by felixfontein**

### CODE REVIEW

**Comment by gotmax23** [comment]
Thanks for cleaning this up! I think we should standardize on `StrPath` so we can use `os.path.join` [^1] and others and not have to worry about converting between str and bytes. It's confusing if some but not all functions support bytes paths.

[^1]: If one argument to `os.path.join` is bytes, all the others have to be.

File: src/antsibull_core/collections.py [hunk 1]
```diff
@@ -9,35 +9,42 @@
...
+        )
+        namespace, collection, _dummy = basename.split("-", 2)
         collection_dir = os.path.join(ansible_collections_dir, namespace, collection)
         # Note: mkdir -p equivalent is okay because we created package_dir ourselves as a directory
         # that only we can access
         os.makedirs(collection_dir, mode=0o700, exist_ok=False)
         installers.append(
             asyncio.create_task(
-                async_log_run(["tar", "-xf", pathname, "-C", collection_dir])
+                async_log_run(["tar", "-xf", f"{pathname}", "-C", collection_dir])
```
Comment (gotmax23) [inline src/antsibull_core/collections.py:39]: It shouldn't be necessary to stringify `pathname`. `async_log_run()` can already handle this.

File: src/antsibull_core/config.py [hunk 2]
```diff
@@ -29,7 +33,7 @@ class ConfigError(Exception):
     pass
 
 
-def find_config_files(conf_files: Iterable[str]) -> list[str]:
+def find_config_files(conf_files: Iterable[StrPath] | Iterable[str]) -> list[str]:
```
Comment (gotmax23) [inline src/antsibull_core/config.py:36]: Does

```suggestion
def find_config_files(conf_files: Iterable[StrPath]) -> list[str]:
```

not work?

**Comment by felixfontein** [comment]
> Thanks for cleaning this up! I think we should standardize on `StrPath` so we can use `os.path.join` [1](#user-content-fn-1-918f88d8be3af276c7464f7b18b69eab) and others and not have to worry about converting between str and bytes. It's confusing if some but not all functions support bytes paths.

We already use `StrOrBytesPath` in multiple places (src/antsibull_core/yaml.py, src/antsibull_core/subprocess_util.py, src/antsibull_core/utils/io.py). At least for hashing I would also use it, since it's very similar to the others. For dependency_files I don't mind switching to `StrPath` (I'll push a commit for that soon).

File: src/antsibull_core/collections.py (hunk 1 above)
Comment (felixfontein) [reply src/antsibull_core/collections.py]: 44445f2fa57956dfcb84535fbfc21edf0efe6fc0.

File: src/antsibull_core/config.py (hunk 2 above)
Comment (felixfontein) [reply src/antsibull_core/config.py]: I tried that first, and for some reason it didn't and I had to use the above. In any case, now it works, don't ask me why :shrug: 2c29d5b2af8762ab81d2156bddf084a94172b6c4

File: src/antsibull_core/config.py [hunk 3]
```diff
@@ -39,21 +43,26 @@ def find_config_files(conf_files: Iterable[str]) -> list[str]:
...
+        os.path.abspath(p)  # pyre-ignore[6]: abspath() accepts path-like object
+        for p in conf_files
+    ]
     flog.fields(paths=paths).info("Paths to check")
 
     config_files = []
     for conf_path in paths:
         if os.path.exists(conf_path):
-            config_files.append(conf_path)
+            config_files.append(f"{conf_path}")
```
Comment (gotmax23) [inline src/antsibull_core/config.py:55]:
```suggestion
            config_files.append(str(conf_path))
```

File: src/antsibull_core/filesystem.py [hunk 4]
```diff
@@ -24,15 +29,15 @@ class CheckFailure(Exception):
...
     """Return the acls for a given file or raise an exception."""
     acls = None
     if posix1e:
-        acl = posix1e.ACL(file=path)
+        acl = posix1e.ACL(file=f"{path}")
         acls = acl.to_any_text(options=posix1e.TEXT_NUMERIC_IDS).decode("utf-8")
     else:
         try:
-            acls = log_run(["getfacl", path, "-n"]).stdout
+            acls = log_run(["getfacl", f"{path}", "-n"]).stdout
```
Comment (gotmax23) [inline src/antsibull_core/filesystem.py:40]:
```suggestion
            acls = log_run(["getfacl", path, "-n"]).stdout
```

File: src/antsibull_core/filesystem.py [hunk 5]
```diff
@@ -24,15 +29,15 @@ class CheckFailure(Exception):
     """We checking failed unexpectedly."""
 
 
-def _get_acls(path: str) -> str:
+def _get_acls(path: StrPath) -> str:
     """Return the acls for a given file or raise an exception."""
     acls = None
     if posix1e:
-        acl = posix1e.ACL(file=path)
+        acl = posix1e.ACL(file=f"{path}")
```
Comment (gotmax23) [inline src/antsibull_core/filesystem.py:36]:
```suggestion
        acl = posix1e.ACL(file=path)
```

File: src/antsibull_core/tarball.py [hunk 6]
```diff
@@ -28,7 +32,8 @@ async def unpack_tarball(tarname: str, destdir: str) -> str:
     :returns: Toplevel of the unpacked directory structure.  This will be
         a subdirectory of `destdir`.
     """
-    manifest = await async_log_run(["tar", "-xzvf", tarname, f"-C{destdir}"])
+    tarname_str = f"{tarname}"
```
Comment (gotmax23) [inline src/antsibull_core/tarball.py:35]:
```suggestion
    tarname_str = str(tarname)
```

**Review by gotmax23 — APPROVED** [review]

**Comment by felixfontein** [comment]
@gotmax23 thanks for reviewing this!

**Pull request merged**